pkg/master/master.go

```diff
@@ -570,16 +570,30 @@ func (m *Master) init(c *Config) {
 	requestInfoResolver := &apiserver.APIRequestInfoResolver{APIPrefixes: sets.NewString(strings.TrimPrefix(defaultVersion.Root, "/")), RestMapper: defaultVersion.Mapper}
 	apiserver.InstallServiceErrorHandler(m.handlerContainer, requestInfoResolver, apiVersions)
 
+	// allAPIVersions records all AVIVersions that are supported at /apis, in the
+	// form of "group/version"
+	allAPIVersions := []string{}
```

Comment:
When we init the third party controller, we should be able to fetch the
versions on the initial run and then remove the package variable in favor
of a member on Master?  If so I'm fine with that as long as you add a TODO
that this should be removed and we have a follow up.

On Tue, Sep 15, 2015 at 2:23 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/master/master.go
> <https://github.com/kubernetes/kubernetes/pull/13955#discussion_r39546584>
> :
>
> > @@ -570,16 +570,30 @@ func (m *Master) init(c *Config) {
> >  	requestInfoResolver := &apiserver.APIRequestInfoResolver{APIPrefixes: sets.NewString(strings.TrimPrefix(defaultVersion.Root, "/")), RestMapper: defaultVersion.Mapper}
> >  	apiserver.InstallServiceErrorHandler(m.handlerContainer, requestInfoResolver, apiVersions)
> >
> > +	// allAPIVersions records all AVIVersions that are supported at /apis, in the
> > +	// form of "group/version"
> > +	allAPIVersions := []string{}
>
> @smarterclayton <https://github.com/smarterclayton> @lavalamp
> <https://github.com/lavalamp>, I move allAPIVersions into init(). When
> Brendan's PR is in, we can pass allAPIVersions to 3rd party controller so
> it can add/delete versions from it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13955/files#r39546584>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift